src/webgl/p5.Geometry.js

```diff
@@ -1369,126 +1369,113 @@ class Geometry {
-        ? this.vertexStrokeColors.slice(
-          currEdge[1] * 4,
-          (currEdge[1] + 1) * 4
-        )
-        : [0, 0, 0, 0];
-      const dir = end
-        .copy()
-        .sub(begin)
-        .normalize();
-      const dirOK = dir.magSq() > 0;
-      if (dirOK) {
-        this._addSegment(begin, end, fromColor, toColor, dir);
-      }
-      if (!this.renderer?._simpleLines) {
-        if (i > 0 && prevEdge[1] === currEdge[0]) {
-          if (!connected.has(currEdge[0])) {
-            connected.add(currEdge[0]);
-            potentialCaps.delete(currEdge[0]);
-            // Add a join if this segment shares a vertex with the previous. Skip
-            // actually adding join vertices if either the previous segment or this
-            // one has a length of 0.
-            //
-            // Don't add a join if the tangents point in the same direction, which
-            // would mean the edges line up exactly, and there is no need for a join.
-            if (lastValidDir && dirOK && dir.dot(lastValidDir) < 1 - 1e-8) {
-              this._addJoin(begin, lastValidDir, dir, fromColor);
-            }
-          }
-        } else {
-          // Start a new line
-          if (dirOK && !connected.has(currEdge[0])) {
-            const existingCap = potentialCaps.get(currEdge[0]);
-            if (existingCap) {
-              this._addJoin(
-                begin,
-                existingCap.dir,
-                dir,
-                fromColor
-              );
-              potentialCaps.delete(currEdge[0]);
-              connected.add(currEdge[0]);
-            } else {
-              potentialCaps.set(currEdge[0], {
-                point: begin,
-                dir: dir.copy().mult(-1),
-                color: fromColor
-              });
-            }
-          }
-          if (lastValidDir && !connected.has(prevEdge[1])) {
-            const existingCap = potentialCaps.get(prevEdge[1]);
-            if (existingCap) {
-              this._addJoin(
-                this.vertices[prevEdge[1]],
-                lastValidDir,
-                existingCap.dir.copy().mult(-1),
-                prevColor
-              );
-              potentialCaps.delete(prevEdge[1]);
-              connected.add(prevEdge[1]);
-            } else {
-              // Close off the last segment with a cap
-              potentialCaps.set(prevEdge[1], {
-                point: this.vertices[prevEdge[1]],
-                dir: lastValidDir,
-                color: prevColor
-              });
-            }
-            lastValidDir = undefined;
-          }
+        const prevEdge = this.edges[i - 1];
+        const currEdge = this.edges[i];
+        const begin = this.vertices[currEdge[0]];
+        const end = this.vertices[currEdge[1]];
+
+        // Copy vertex properties into buffers for stroke vertices
+        if (this.vertexProperties) {
+            this.vertexProperties.push([...this.vertexProperties[currEdge[0]]]);
```

Comment:
I think we might need to do this more than once, depending on what we add. So I think every edge has 4 vertices (we render segments as rectangles basically), so we'd have to double these up. But also, between some segments we have quads for joins, and at some segments we also have caps, and all of the vertices for those need vertex properties too, otherwise we won't have enough data to match all the vertices.

`_addCap`, `_addJoin`, and `_addSegment` already do this duplication for positions and colors, so those methods may need to be modified to pass in vertex properties so that they can duplicate and add them too.